### ISSUE TITLE
"& has lower precedence than ==; == will be evaluated first" in AliAnalysisTaskCFTree.cxx

### ISSUE THREAD

**Issue by cbourjau** [alisw/AliPhysics#2351]
Hello,

I was just trying to find some unrelated information in AliAnalysisTaskCFTree.cxx when my static analyzer lint pointed out a (likely) logic bug to me. I will add more information with a comment directly to the code in a moment. 

I don't use this code myself but it seems to have been edited by @qgp, @ekryshen, @aohlson. I thought I would just point this out here.

By the way, this is a warning with -Wall, as recently discussed at #2213

**Comment by cbourjau** [comment]
https://github.com/alisw/AliPhysics/blob/a64a25f1ad39d20841ed76975436b4f0f08e84ea/PWGCF/Correlations/Base/AliAnalysisTaskCFTree.cxx#L578

```cpp

if (fTrackFilterBit != 0) {
    // The == is evaluated first, but we just checked that its not 0
    // So this just compares the mask to `false`
    // I guess thats not what its supposed to do?
    if (mask & fTrackFilterBit == 0)
        continue;
}
```

**Comment by akubera** [comment]
Fixed by Hans in #2192; this issue can be closed.
